### PR TITLE
Add Terraform workspace support

### PR DESCRIPTION
**Pull request by marcinwyszynski**
## Description of the change

This change adds support for setting Terraform workspace, and factors out Terraform settings into a separate block (optional for now).

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Pull Request is no longer marked as "draft"
- [x] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: spacelift/resource_stack.go [hunk 1]
```diff
@@ -397,10 +439,20 @@ func stackInput(d *schema.ResourceData) structs.StackInput {
 				StackName: toString(pulumi[0].(map[string]interface{})["stack_name"]),
 			},
 		}
-	} else {
-		ret.VendorConfig = &structs.VendorConfigInput{
-			Terraform: &struct{}{},
+	} else if terraform, ok := d.Get("terraform").([]interface{}); ok && len(terraform) > 0 {
```
Comment (cube2222) [inline spacelift/resource_stack.go:442]: I can't comment on this, but don't we want the above 
```
if terraformVersion, ok := d.GetOk("terraform_version"); ok {
		ret.TerraformVersion = toOptionalString(terraformVersion)
	}
```

to set the vendor specific config?

Comment (cube2222) [reply spacelift/resource_stack.go:442]: So we can get rid of the top-level one from the API

**Review by cube2222 — APPROVED** [review]

**Review by cube2222 — APPROVED** [review]
With nit

**Review by b4k3r — APPROVED** [review]

File: spacelift/resource_stack.go (hunk 1 above)
Comment (marcinwyszynski) [reply spacelift/resource_stack.go:444]: Done, I think.

**Pull request merged**